spec/functional/resource/dpkg_package_spec.rb

```diff
@@ -45,11 +45,17 @@ def set_dpkg_package_name(name)
     dpkg_package.package_name name
   end
 
-  def should_be_purged_or_removed(package, action)
+  def should_be_purged_or_removed(package, action=nil)
     if action == :purge
-      shell_out!("dpkg -s #{package}", returns: [1])
+        expect(shell_out(
+          'dpkg -s chef-integration-test').stdout).to match(/not-installed|no info/)
```

Comment:
so this is where you want to use something like `.not_to match(/deinstall|installed/)` -- or else you need to invert that and test for either exit(1) or `match(/not-installed/)`.   This is the 'i want to know if the package is really not installed' clause.  You could also make this the default, since this is what you're wanting to test in the 'nil' case as well.